### ISSUE TITLE
--offline still creates an isolated <project_dir>/.ansible cache, polluting editor workspaces

### ISSUE THREAD

**Issue by bdsoha** [ansible/ansible-lint#5050]
### Summary

Even with `--offline`, ansible-lint creates an isolated cache directory `<project_dir>/.ansible/` (containing `collections/`, `roles/`, `modules/`) at the project root on every real lint run. The project root resolves to the git top-level, so a stray, un-git-ignored directory is dropped into the repo. The Ansible VS Code extension (`redhat.ansible`) invokes `ansible-lint --offline ...` on every file open/save, so every editor user hits this. In `--offline` mode nothing is installed, so the isolated project cache provides no benefit.

Follow-up to #4533 (PR #4894 only suppressed creation for trivial invocations such as `--version`); relates to #4501 and ansible/vscode-ansible#1922.

### Issue Type

Bug Report

### OS / Environment

```
ansible-lint 26.4.0 using ansible-core:2.20.5 ansible-compat:26.3.0 ruamel-yaml:0.19.1
```
Installed via `uv tool`, Linux.

### Steps to reproduce

```bash
cd /path/to/any/git/repo/with/playbooks
rm -rf .ansible
ansible-lint --offline --nocolor -f codeclimate playbook.yml
ls .ansible        # collections/ roles/ modules/ now exist at the repo root
```
(Those flags are exactly what the `redhat.ansible` language server passes.)

### Desired behavior

With `--offline`, ansible-lint should not create an isolated `<project_dir>/.ansible` cache - or should place it in a non-project, git-invisible location (e.g. under `XDG_CACHE_HOME`), or expose a documented setting to relocate it (cf. #4501). Offline runs install nothing, so the per-project isolated cache has no purpose in that mode. In #4501 @ssbarnea noted "if offline mode is enabled, we could argue that we should never attempt to install things."

### Actual behavior

`.ansible/` is created at the git project root on every lint run. Root cause, from the installed source:

- `ansiblelint/app.py` constructs `ansible_compat.Runtime(project_dir=..., isolated=True)` - `isolated=True` is hardcoded; no CLI flag, config key, or env var changes it. `ANSIBLE_HOME` is consulted only in the `isolated=False` branch; `VIRTUAL_ENV` does not help because the project-dir probe runs unconditionally first.
- `project_dir` is resolved by `find_project_root()` to the first `.git` ancestor - the repo root.
- `ansible_compat.prerun.get_cache_dir(project_dir, isolated=True)` sets the cache to `project_dir/.ansible` and calls `is_writable()` on it - and `is_writable()` itself does `path.mkdir(parents=True, exist_ok=True)`. The directory is created merely by the writability probe, before any linting or any install occurs. `--offline` does not gate this.

Users currently have to add `.ansible/` to `.gitignore` per-project as a workaround.

**Comment by sibebleuze** [comment]
Thank you for creating this. I just wanted to add here what I said before (in one of the tickets mentioned above): there is already a fallback option [in the code](https://github.com/ansible/ansible-compat/blob/main/src/ansible_compat/prerun.py#L83) which puts the .ansible folder in some temporary folder.
This issue could be fixed by something as simple as an ansible-lint configuration option / CLI flag / ... that skips all the other options and jumps straight to that temporary folder option.
Alternatively, a "clean up after yourself" option where the .ansible folder is removed again afterwards could work as well (if created by the current ansible-lint execution of course, wouldn't want to delete anything that was already there).

Lastly, multiple of the (closed) issues mentioned in the post indicate the [docs](https://docs.ansible.com/projects/lint/usage/#caching) are out of date and mention a .cache folder instead of a .ansible folder.

**Comment by sibebleuze** [comment]
@rockygeekz Could you indicate which PR closed this issue? Or which ansible-lint version the fix was released in?
@Himanshuagrawal4 you mention in https://github.com/ansible/ansible-lint/pull/5063 that the docs issue remains tracked here, but the issue is now closed so that is no longer the case. Perhaps it should be tracked in a separate issue?

**Comment by Himanshuagrawal4** [comment]
You're right. When I closed #5063, #5050 was still open; it was closed later without a linked fixing PR or release. I checked the current main branch, and docs/usage.md still describes the project cache as {project_dir}/.cache, so the documentation gap remains. I also cannot identify from this thread which release, if any, fixed the runtime behavior. A separate documentation issue (or reopening the documentation part here) would be appropriate.

**Comment by sibebleuze** [comment]
I also don't see which PR/commit changed this, but I did test and I can confirm that in the latest release (26.8.0) there is no `{project_dir}/.ansible` folder created by `ansible-lint --offline`. Executing `ansible-lint` without the offline flag does still create it, but I guess that is intended then (even though the resulting folders are still empty for me like they were in https://github.com/ansible/ansible-lint/issues/4533#issuecomment-3725217176)?

I suppose that leaves you to create a separate issue for the docs.

**Comment by Himanshuagrawal4** [comment]
Thanks for testing that. I found #3371, which is already open for the caching documentation mismatch. I've added the current 26.8.0 behavior there instead of opening a duplicate. That should keep the runtime fix tracked here separate from the documentation follow-up.
